File: find-the-array-concatenation-value/Solution2562.java
// https://leetcode.com/problems/find-the-array-concatenation-value/

class Solution2562 {

    // T: O(n), S: O(1)
    public long findTheArrayConcVal(int[] nums) {
        long ans = 0L;
        for (int left = 0, right = nums.length - 1; left <= right; left++, right--) {
            var numStr = "";
            if (left != right) {
                numStr = String.valueOf(nums[left]) + String.valueOf(nums[right]);
            } else {
                numStr = String.valueOf(nums[left]);
            }
            ans += Long.parseLong(numStr);
        }
        return ans;
    }

    public static void main(String[] args) {
        var sol = new Solution2562();
        var nums1 = new int[] { 7, 52, 2, 4 };
        var nums2 = new int[] { 5, 14, 13, 8, 12 };
        System.out.println(sol.findTheArrayConcVal(nums1)); // 596
        System.out.println(sol.findTheArrayConcVal(nums2)); // 673
    }

}

File: find-the-array-concatenation-value/solution2562.go
// https://leetcode.com/problems/find-the-array-concatenation-value/

package main

import (
	"fmt"
	"strconv"
)

// T: O(n), S: O(1)
func findTheArrayConcVal(nums []int) int64 {
	var ans int64
	for i, j := 0, len(nums)-1; i <= j; i, j = i+1, j-1 {
		var numStr string
		if i != j {
			numStr = strconv.Itoa(nums[i]) + strconv.Itoa(nums[j])
		} else {
			numStr = strconv.Itoa(nums[i])
		}
		num, _ := strconv.ParseInt(numStr, 10, 64)
		ans += num
	}
	return ans
}

func main() {
	nums1 := []int{7, 52, 2, 4}
	nums2 := []int{5, 14, 13, 8, 12}
	fmt.Println(findTheArrayConcVal(nums1)) // 596
	fmt.Println(findTheArrayConcVal(nums2)) // 673
}
